libmbim: 1.24.8 -> 1.26.0

flokli opened the pull request:
```
Overview of changes in libmbim 1.26
----------------------------------------

 * Build now requires GLib/GObject/GIO 2.56.

 * The GUdev optional build/runtime requirement is now fully dropped, it's no
   longer used.

 * Building from git no longer requires autoconf-archive, the needed AX_ macros
   are now shipped inside m4/.

 * In addition to building from a source release tarball, or building from git
   checkouts using the GNU autotools suite (autoconf/automake/libtool), this
   release includes the initial support for the meson build system. The meson
   port is not fully complete yet, as there are some missing things in the doc
   generation steps, but for system integration or development purposes, the
   port should be fully operational. This major release, including all its
   stable updates in the 1.26.x series, will be the last ones providing support
   for GNU autotools. The next major release will likely be a meson-only one,
   and will therefore not be based on a source release tarball any more, but
   on specific git tags instead.

 * Implemented new link management operations, exclusively for the cdc_mbim
   driver for now. These new operations allow creating or deleting VLAN network
   interfaces in order to run multiplexed data sessions over one single physical
   network interface.

 * Added support for the Microsoft-defined SAR service, including the following
   operations:
   ** MBIM_CID_MS_SAR_CONFIG
   ** MBIM_CID_MS_SAR_TRANSMISSION_STATUS

 * Added support for the Microsoft-defined UICC Low Level Access service,
   including the following operations:
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_ATR
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_OPEN_CHANNEL
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_CLOSE_CHANNEL
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_APDU
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_TERMINAL_CAPABILITY
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_RESET

 * Added support for the Qualcomm-defined QDU service, including the following
   operations:
   ** MBIM_CID_QDU_UPDATE_SESSION
   ** MBIM_CID_QDU_FILE_OPEN
   ** MBIM_CID_QDU_FILE_WRITE

 * Extended the Microsoft-defined Basic Connect Extensions service, including
   the following operations:
   ** MBIM_CID_MS_BASIC_CONNECT_EXTENSIONS_DEVICE_CAPS
   ** MBIM_CID_MS_BASIC_CONNECT_EXTENSIONS_SYS_CAPS
   ** MBIM_CID_MS_BASIC_CONNECT_EXTENSIONS_SLOT_INFO_STATUS
   ** MBIM_CID_MS_BASIC_CONNECT_EXTENSIONS_DEVICE_SLOT_MAPPINGS

 * libmbim-glib:
   ** Logic updated to make sure full packets are written at once, instead of
      writing them in chunks.
   ** Updated the "LTE attach status" APIs in order to avoid creating unneeded
      struct types in the interface. The older methods have been deprecated and
      maintained in the library for compatibility purposes only.

 * mbim-proxy:
   ** Internal buffer size updated from 512 bytes to 4096 bytes.

 * mbimcli:
   ** New '--ms-set-sar-config' and '--ms-query-sar-config' actions.
   ** New '--ms-set-transmission-status' and '--ms-query-transmission-status'
      actions.
   ** Updated '--enter-pin', '--disable-pin' and '--unlock-pin' to allow other
      PIN types, not just assuming PIN1.
   ** New '--link-add', '--link-delete', '--link-list' and '--link-delete-all'
      actions.
   ** New '--ms-query-sys-caps' action.
   ** New '--ms-query-slot-info-status' action.
   ** New '--ms-query-device-slot-mappings' and '--ms-set-device-slot-mappings'
      actions.
   ** Renamed '--ms-query-lte-attach-status' to '--ms-query-lte-attach-info',
      and kept the old name for compatibility purposes.

 * mbim-network:
   ** When using the mbim-proxy, skip trying to manage the MBIM session and
      transaction ids as that is implicitly done by the proxy already.

 * Several other minor improvements and fixes.

The following features which were backported to 1.24.x releases are also present
in libmbim 1.26.0:

 * Fixed merged subscribe list reporting and handling in the proxy.
 * Fixed transaction id handling when multiple fragments are involved.
 * Fixed read overflows on malformed messages.
 * Skip warnings if descriptors file cannot be read, as in new MBIM backends
   other than cdc_mbim.
```

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Ma27 committed:
nixos/captive-browser: fix startup

It seems as since Chromium 92, `chromium` crashes on startup if
`XDG_CONFIG_HOME` points to a read-only (store-)path.

Ma27 committed:
captive-browser: 2019-04-16 -> 2021-08-01

Changes: https://github.com/FiloSottile/captive-browser/compare/08450562e58bf9564ee98ad64ef7b2800e53338f...9c707dc32afc6e4146e19b43a3406329c64b6f3c

lheckemann committed:
Merge pull request #132338 from Ma27/fix-captive-browser-startup

nixos/captive-browser: fix startup

lheckemann committed:
Merge pull request #132353 from Ma27/bump-captive-browser

captive-browser: 2019-04-16 -> 2021-08-01

flokli committed:
libmbim: 1.24.8 -> 1.26.0

Overview of changes in libmbim 1.26
----------------------------------------

 * Build now requires GLib/GObject/GIO 2.56.

 * The GUdev optional build/runtime requirement is now fully dropped, it's no
   longer used.

 * Building from git no longer requires autoconf-archive, the needed AX_ macros
   are now shipped inside m4/.

 * In addition to building from a source release tarball, or building from git
   checkouts using the GNU autotools suite (autoconf/automake/libtool), this
   release includes the initial support for the meson build system. The meson
   port is not fully complete yet, as there are some missing things in the doc
   generation steps, but for system integration or development purposes, the
   port should be fully operational. This major release, including all its
   stable updates in the 1.26.x series, will be the last ones providing support
   for GNU autotools. The next major release will likely be a meson-only one,
   and will therefore not be based on a source release tarball any more, but
   on specific git tags instead.

 * Implemented new link management operations, exclusively for the cdc_mbim
   driver for now. These new operations allow creating or deleting VLAN network
   interfaces in order to run multiplexed data sessions over one single physical
   network interface.

 * Added support for the Microsoft-defined SAR service, including the following
   operations:
   ** MBIM_CID_MS_SAR_CONFIG
   ** MBIM_CID_MS_SAR_TRANSMISSION_STATUS

 * Added support for the Microsoft-defined UICC Low Level Access service,
   including the following operations:
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_ATR
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_OPEN_CHANNEL
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_CLOSE_CHANNEL
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_APDU
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_TERMINAL_CAPABILITY
   ** MBIM_CID_MS_UICC_LOW_LEVEL_ACCESS_RESET

 * Added support for the Qualcomm-defined QDU service, including the following
   operations:
   ** MBIM_CID_QDU_UPDATE_SESSION
   ** MBIM_CID_QDU_FILE_OPEN
   ** MBIM_CID_QDU_FILE_WRITE

 * Extended the Microsoft-defined Basic Connect Extensions service, including
   the following operations:
   ** MBIM_CID_MS_BASIC_CONNECT_EXTENSIONS_DEVICE_CAPS
   ** MBIM_CID_MS_BASIC_CONNECT_EXTENSIONS_SYS_CAPS
   ** MBIM_CID_MS_BASIC_CONNECT_EXTENSIONS_SLOT_INFO_STATUS
   ** MBIM_CID_MS_BASIC_CONNECT_EXTENSIONS_DEVICE_SLOT_MAPPINGS

 * libmbim-glib:
   ** Logic updated to make sure full packets are written at once, instead of
      writing them in chunks.
   ** Updated the "LTE attach status" APIs in order to avoid creating unneeded
      struct types in the interface. The older methods have been deprecated and
      maintained in the library for compatibility purposes only.

 * mbim-proxy:
   ** Internal buffer size updated from 512 bytes to 4096 bytes.

 * mbimcli:
   ** New '--ms-set-sar-config' and '--ms-query-sar-config' actions.
   ** New '--ms-set-transmission-status' and '--ms-query-transmission-status'
      actions.
   ** Updated '--enter-pin', '--disable-pin' and '--unlock-pin' to allow other
      PIN types, not just assuming PIN1.
   ** New '--link-add', '--link-delete', '--link-list' and '--link-delete-all'
      actions.
   ** New '--ms-query-sys-caps' action.
   ** New '--ms-query-slot-info-status' action.
   ** New '--ms-query-device-slot-mappings' and '--ms-set-device-slot-mappings'
      actions.
   ** Renamed '--ms-query-lte-attach-status' to '--ms-query-lte-attach-info',
      and kept the old name for compatibility purposes.

 * mbim-network:
   ** When using the mbim-proxy, skip trying to manage the MBIM session and
      transaction ids as that is implicitly done by the proxy already.

 * Several other minor improvements and fixes.

The following features which were backported to 1.24.x releases are also present
in libmbim 1.26.0:

 * Fixed merged subscribe list reporting and handling in the proxy.
 * Fixed transaction id handling when multiple fragments are involved.
 * Fixed read overflows on malformed messages.
 * Skip warnings if descriptors file cannot be read, as in new MBIM backends
   other than cdc_mbim.

lukegb commented on pkgs/development/libraries/libmbim/default.nix:
```diff
@@ -11,11 +12,11 @@
-    sha256 = "sha256-AlkHNhY//xDlcyGR/MwbmSCWlhbdxZYToAMFKhFqPCU=";
+    sha256 = "1kqkx139z62w391bz6lwmcjg7v12jxlcm7hj88222xrcn8k0j7qy";
   };
 
```
Is the `libgudev` buildInput removable now, per the change notes?

flokli replied:
Right, I missed that part. Removed it, and confirmed it still builds.

lukegb commented on pkgs/development/libraries/libmbim/default.nix:
```diff
@@ -11,11 +12,11 @@
+    sha256 = "1kqkx139z62w391bz6lwmcjg7v12jxlcm7hj88222xrcn8k0j7qy";
   };
 
   outputs = [ "out" "dev" "man" ];
```
Do we want to switch from autotools to meson/ninja now, or wait for the first meson-only release? (Apparently some doc-generation steps are missing judging by the readme so we might want to wait, I guess)

flokli replied:
Yeah, I'd like to wait.

lukegb reviewed (approved):
Might be worth updating the license to gpl2Plus while you're here (https://gitlab.freedesktop.org/mobile-broadband/libmbim/-/blob/master/meson.build#L7) but otherwise LGTM

flokli committed:
libmbim: update license to gpl2Plus

As per
https://gitlab.freedesktop.org/mobile-broadband/libmbim/-/blob/master/meson.build#L7.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 132447` at 38cdd5fc run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/4f160f5752385d533837091b20f47e47">libsForQt512.modemmanager-qt</a></li>
  </ul>
</details>
<details>
  <summary>388 packages skipped due to time constraints:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>alfis</li>
    <li>almanah</li>
    <li>apostrophe</li>
    <li>appgate-sdp</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>astroid</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>15 packages built successfully:</summary>
  <ul>
    <li>geoclue2</li>
    <li>glib-networking (gnome.glib-networking ,gnome.glib_networking)</li>
    <li>gnome.gnome-settings-daemon (gnome.gnome_settings_daemon)</li>
    <li>gnome.mutter</li>
    <li>networkmanagerapplet (gnome.networkmanagerapplet)</li>
    <li>libmbim</li>
    <li>libnma</li>
    <li>libproxy</li>
    <li>libqmi</li>
    <li>libsForQt5.modemmanager-qt (libsForQt515.modemmanager-qt ,plasma5Packages.modemmanager-qt)</li>
    <li>libsForQt514.modemmanager-qt</li>
    <li>modemmanager</li>
    <li>networkmanager</li>
    <li>pantheon.elementary-settings-daemon</li>
    <li>purple-mm-sms</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

flokli commented:
Changed the base to `staging`, as this rebuilds a lot (due to it being pulled in by modemmanager -> networkmanager)
